### PR TITLE
Main README converted to Markdown

### PR DESCRIPTION
**Pull request by v01d**
## Summary

As discussed on the list, here's the main README converted to Markdown. If you go to "view file" on GitHub, it should be rendered as GitHub renders Markdown files. This was a mostly mechanical conversion of the syntax. I probably missed some places where I should use backticks for commands, files, etc. mentioned in the text. I can do that in a second pass.
Have a look and feel free to point out places where I missed something.

PS: maybe you can cancel the CI when it triggers since it will be unnecessary for this PR.

## Impact

Nothing, just a README

## Testing

Proofread the formatting in GitHub and on own Markdown editor to confirm it looks as intended.

### CODE REVIEW

**Comment by Ouss4** [comment]
@btashton @adamfeuer @w8jcik Would you like to take a look?

**Comment by adamfeuer** [comment]
@Ouss4 I will check it out and report back. 🙂

**Comment by adamfeuer** [comment]
@Ouss4 @v01d Overall it looks great, the rendering on Github is a great help. I suggested some minor formatting changes.

**Comment by w8jcik** [comment]
@v01d This is one big readme :) I will try to help if you guys won't be done soon. What is visible for me, right now:
- You should use `git mv` on this file, otherwise the history is lost.
- You can use backticks with titles as well. They look fine and are more readable I think.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,2513 @@
...
+
+**UPDATE**:  Sergey Frolov had success with the following minimal
+Cygwin configuration:
+   
+   1. After starting the Cygwin installer, keep the recommended
+      packages that are pre-selected in the default configuration.
+   
+   2. Using the installation tools, add the following packages:
+      
+        make (GNU make)  bison        libgmp3-dev
```
[the remark this answers is not in the document]
Comment (v01d) [reply README.md:283]: the indentation appears to be relative to the enumeration, so it is actually missing two spaces. at least doing that i can see it looks like it should.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,2513 @@
...
+        style delimiters.  For example, change "..\apps" to "../apps"
+     
+     And of course, after you use the configuration tool you need to
+     restore CONFIG_WINDOWS_NATIVE=y and the correct CONFIG_APPS_DIR.
+
+  2. You can, with some effort, run the Cygwin kconfig-mconf tool
+     directly in the Windows console window.  In this case, you do not
+     have to modify the .config file, but there are other complexities:
+     
+    a. You need to temporarily set the Cygwin directories in the PATH
```
[the remark this answers is not in the document]
Comment (v01d) [reply README.md:1347]: I'm previewing the result in different Markdown editors and it looks OK to me.

**Comment by v01d** [comment]
I added all suggested changes (mostly a lot of backticks).

**Comment by xiaoxiang781216** [comment]
@v01d is it ready for merging?

**Comment by v01d** [comment]
Yes, looks OK to me.

**Comment by v01d** [comment]
If you consider history important I can try to make it so it is a rename, but I'm not really sure how to do so safely.

**Comment by adamfeuer** [comment]
@v01d One way to fix this branch to preserve README history would be the following:

1. locally rename `v01d:readme` to `v01d:readme-old`  
1. make a new branch from a fresh copy of master calling it `v01d:readme`
1. do the `git mv` and commit it
1. `git merge --squash v01d:readme-old` – this will take the changes from the old branch as a change into the new branch
1. commit the change
1. force push the branch here

That should fix the history.

**Comment by v01d** [comment]
I tried with a rebase and with your approach but I get a conflict:
<pre>
CONFLICT (rename/delete): README.txt deleted in readme and renamed to README.md in HEAD. Version HEAD of README.md left in tree.
Squash commit -- not updating HEAD
Automatic merge failed; fix conflicts and then commit the result.
</pre>

and README.md is left with the original contents of the .txt. I was trying to look if using rebase I could only select the added changes to the file and not the deletion of the .txt since that is what mostly conflicts with the git mv.

**Comment by adamfeuer** [comment]
@v01d I'll try it myself and see if I can come up with a better way. 😄

**Comment by v01d** [comment]
Ok, did it the manual way: create a branch, made the rename and then used `git show readme:README.md > README.md`.
Should be all set now.

**Comment by adamfeuer** [comment]
@v01d Cool. That's similar to what I just worked out.

**Comment by v01d** [comment]
@xiaoxiang781216 you can merge now when you're able

**Comment by xiaoxiang781216** [comment]
@v01d done.

**Pull request merged**